### DISCUSSION TITLE
Detect when Vector is lagging

### DISCUSSION THREAD

**Discussion by ShahroZafar** [vectordotdev/vector#20432] [Q&A]
Is there a way using which we can somehow detect by how much the vector is falling behind reading events from a particular file?

Currently we want to use vector to use file source to push logs of a very critical application (Almost 0 log loss) to kafka and we want to get alerted if vector is lagging behind and is not keeping up with the load.
The deployment is in kubernetes (We are currently not using kubernetes_logs source as log enrichment doesn't matter and we want to keep the log entry size as small as possible)

**Comment by jszwedko** [comment]
My suggestion for this would be to add a timestamp to each event as it is written to the file and then you could measure the difference between the timestamp being read and "now" (in Vector this could be done using a `remap` transform to calculate the difference and then the `log_to_metric` transform to turn that into a metric).

**Reply by ShahroZafar** [reply]
Right. I will give it a try. Thanks

Also is there a way to get information about how many files are actively watched? I mean a way using which we can be sure that vector has not missed / skipped reading a file because the rotation is very very fast (due to load)
